lib/widgets/containers/tabs/dashboard_tab_widgets.dart

```diff
@@ -0,0 +1,278 @@
+      ],
+    );
+  }
+}
+
+class PegOutAction extends StatelessWidget {
+  const PegOutAction({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ViewModelBuilder.reactive(
+      viewModelBuilder: () => PegOutViewModel(),
+      builder: ((context, viewModel, child) {
+        return DashboardActionModal(
+          'Peg-out to mainchain',
+          endActionButton: SailButton.primary(
+            label: 'Execute peg-out',
+            loading: viewModel.isBusy,
+            size: ButtonSize.small,
+            onPressed: () async {
+              viewModel.executePegOut();
+            },
+          ),
+          children: [
+            LargeEmbeddedInput(
+              controller: viewModel.bitcoinAddressController,
+              hintText: 'Enter a mainchain bitcoin-address',
+            ),
+            LargeEmbeddedInput(
+              controller: viewModel.bitcoinAmountController,
+              hintText: 'Enter a BTC-amount',
+              suffixText: 'BTC',
+              bitcoinInput: true,
+            ),
+            const StaticActionField(
+              label: 'Mainchain fee',
+              value: 'TBD BTC',
+            ),
+            const StaticActionField(
+              label: 'Sidechain fee',
+              value: 'TBD BTC',
+            ),
+            StaticActionField(
+              label: 'Total amount',
+              value: '${viewModel.totalBitcoinAmount} BTC',
+            ),
+          ],
+        );
+      }),
+    );
+  }
+}
+
+class PegOutViewModel extends BaseViewModel {
+  final log = Logger(level: Level.debug);
+  AppRouter get _router => GetIt.I.get<AppRouter>();
+
+  final bitcoinAddressController = TextEditingController();
+  final bitcoinAmountController = TextEditingController();
+  String get totalBitcoinAmount => (double.tryParse(bitcoinAmountController.text) ?? 0).toStringAsFixed(8);
+
+  PegOutViewModel() {
+    bitcoinAddressController.addListener(notifyListeners);
+    bitcoinAmountController.addListener(notifyListeners);
+  }
+
+  void executePegOut() async {
+    setBusy(true);
+    notifyListeners();
+    await Future.delayed(const Duration(seconds: 1), () {});
+
+    setBusy(false);
+    notifyListeners();
+
+    await _router.pop();
+  }
+}
+
+class PegInAction extends StatelessWidget {
+  const PegInAction({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ViewModelBuilder.reactive(
+      viewModelBuilder: () => PegInViewModel(),
+      builder: ((context, viewModel, child) {
+        return DashboardActionModal(
+          'Peg-in to mainchain',
+          endActionButton: SailButton.primary(
+            label: 'Generate new address',
+            loading: viewModel.isBusy,
+            size: ButtonSize.small,
+            onPressed: () async {
+              viewModel.generateNewAddress();
+            },
+          ),
+          children: const [
+            StaticActionField(
```

Comment:
Make the `value` field here copyable on click?

Reply:
yes, definitely